saleor/graphql/discount/mutations/promotion/promotion_create.py

```diff
@@ -183,7 +183,8 @@ def _save_m2m(
             models.PromotionRule.objects.bulk_create(rules)
 
         for rule, channels in rules_with_channels_to_add:
-            rule.channels.set(channels)
+            if channels:
```

Comment:
Name `rules_with_channels_to_add` suggest me that I have a `rule` with `channels`. But it looks like, there is a possibility that `channels` will be None. 
So my question is: why we add rule without a channel to `rules_with_channels_to_add` ? For me it looks like this `if` should be place between lines: 180 and 181.
I know, that this PR is port of changes, but maybe we could clean it up on main at least.

Reply:
https://github.com/saleor/saleor/pull/15260